### ISSUE TITLE
Missing analysis warning for const loop variables?

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#32038]
While working on flutter I wrote some code that gave no warnings but crashed at runtime with this:

```text
_CompileTimeError: 'package:flutter_tools/src/android/vscode.dart': error: line 62: Loop variable cannot be 'const'
    for (const String root in["/", homeDirPath]) {
```

I tried it in DartPad and saw the same:

https://dartpad.dartlang.org/144509325db6780e8b6de13065399429

If this code is invalid I would've expected a warning/error at dev time (maybe this would be fixed in the CFE? @bwilkerson @devoncarew)

**Comment by bwilkerson** [comment]
Yes, it appears that analyzer does not produce the required error here.
